### PR TITLE
Add additional account information

### LINKED ISSUE #27: Add additional account information
- [x] username
- [x] firstname, lastname
- [x] place of residence
- [x] birth date

### PR DESCRIPTION
**Pull request by felixauringer**
Closes #27

### CODE REVIEW

**Review by konradh — APPROVED** [review]

**Review by mpass99 — APPROVED** [review]

File: spec/features/user_registration_spec.rb [hunk 1]
```diff
@@ -1,13 +1,16 @@
 require 'rails_helper'
 
 RSpec.describe "Users", type: :feature do
-  it "can sign up" do
+
```
Comment (felixauringer) [inline spec/features/user_registration_spec.rb:4]: Rubocop does not like this empty line.

Comment (chrisma) [reply spec/features/user_registration_spec.rb:4]: If you disagree with RuboCop, feel free to edit its config. Tools should not burden you!

File: spec/features/user_registration_spec.rb [hunk 2]
```diff
@@ -21,4 +24,17 @@
...
+    within '#new_user' do
+      user_attrs = FactoryBot.attributes_for(:user)
+      fill_in "Email", with: user_attrs[:email]
+      fill_in "Password", with: user_attrs[:password]
+      fill_in "Password confirmation", with: "wrong_password"
+      find('input[type="submit"]').click
+    end
+
+    expect(page).to have_css(".alert-danger", count: 1)
+    expect(page).to have_content("prohibited this user")
```
Comment (chrisma) [inline spec/features/user_registration_spec.rb:38]: Rely on translation file? E.g. https://github.com/hpi-swt2/connections-portal/blob/dev/config/locales/devise.en.yml#L65

File: spec/features/user_registration_spec.rb (hunk 1 above)
Comment (LinqLover) [reply spec/features/user_registration_spec.rb:4]: I like RuboCop for asserting a common coding style. Do we have any format to discuss questions like this in the large group?

File: spec/models/user_spec.rb [hunk 3]
```diff
@@ -9,13 +9,39 @@
...
     expect(@user).not_to be_valid
   end
+
+  it "is not valid without a username" do
+    @user.username = ""
+    expect(@user).not_to be_valid
+  end
+
+  it "creates user name from email" do
+    user = described_class.new({ email: "test-user@example.org" })
```
Comment (chrisma) [inline spec/models/user_spec.rb:39]: Or use a factory and specify the `email` attribute? https://github.com/hpi-swt2/connections-portal/blob/dev/spec/factories/users.rb

File: spec/features/user_registration_spec.rb (hunk 1 above)
Comment (chrisma) [reply spec/features/user_registration_spec.rb:4]: Maybe propose changes you'd like to make to the `.rubocop.yml` in a PR? Then that could be used for discussion on concrete changes

Comment (chrisma) [reply spec/features/user_registration_spec.rb:4]: For example, I would probably disable/modify the rules that lead to this: 
image

Comment (chrisma) [reply spec/features/user_registration_spec.rb:4]: I would suggest the following for `.rubocop.yml`. Empty lines might be nice for structuring tests?

```Ruby
Layout/EmptyLinesAroundBlockBody:
  Exclude:
    - 'spec/**/*.rb'
```

**Comment by frcroth** [comment]
Moved to #52

**Pull request closed without merging**